preupg/xmlgen/compose.py

```diff
@@ -110,12 +146,12 @@ def collect_group_xmls(source_dir, content=None, level=0,
                 continue
             try:
                 ret[dirname] = (ElementTree.parse(group_file_path).getroot(),
-                                ComposeXML.collect_group_xmls(new_dir,
-                                level=level + 1,
-                                generate_from_ini=generate_from_ini))
+                                ComposeXML.collect_group_xmls(
+                                    module_set_dir, new_dir,
+                                    generate_from_ini=generate_from_ini))
             except ParseError as e:
-                print ("Encountered a parse error in file ", group_file_path,
-                       " details: ", e)
+                print("Encountered a parse error in file ", group_file_path,
+                      " details: ", e)
```

Comment:
Use log_message function for this error message, with level=logging.ERROR. Then it should stop execution. I think sys.exit(1) can be sufficient.